apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -65,22 +90,28 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
                   )
                   .map(([name]) => name);
 
-    useEffect(() => {
-        const newSearchParams = new URLSearchParams(searchParams);
-        newSearchParams.set('module', selectedModule);
-        setSearchParams(newSearchParams);
-    }, [selectedModule, setSearchParams, searchParams]);
-
     const submitSearch = useCallback(() => {
-        setSelectedModule((prev: string) =>
-            filteredModules.length === 1 ? filteredModules[0] : prev
-        );
-    }, [filteredModules]);
+        if (filteredModules.length === 1)
+            setModule(searchParams, setSearchParams, filteredModules[0]);
+    }, [filteredModules, searchParams, setSearchParams]);
+
+    const onChangeModule = useCallback(
+        (newModule: string) => () =>
+            setModule(searchParams, setSearchParams, newModule),
+        [searchParams, setSearchParams]
+    );
 
     return (
         <div className="flex flex-col md:flex-row md:flex-nowrap gap-5 border-0 border-y border-solid border-sui-grey-45">
             <div className="w-full md:w-1/5">
-                <Combobox value={selectedModule} onChange={setSelectedModule}>
+                <Combobox
+                    value={getModule(
```

Comment:
Rather than doing this online every, just make a variable that computes this at the top of the component and reuse it